### ISSUE TITLE
Rule blocked: CORERULES-9641 - missing values reported as "null" string value in output variables

### ISSUE THREAD

**Issue by ASL-rmarshall** [cdisc-org/cdisc-rules-engine#1159]
- JSON containing the full request: [Request.json](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeERoQ1E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c69ed1db5e17509c18d9d54a5cf84611f965b09b/Request.txt)

**Links to related JIRA Tickets**
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-9641

**Rule Information**
- **Standard**: USDM
- **Rule ID**: DDF00098
- **Rule Description**: 

**Describe the bug**
Previously, when rule output contained empty/missing/null values (either because the input variable was empty or because a matched record was not available in a left join), these were reported in the output variables as `null`.  Now they are being reported as `"null"`, which implies that the specified variable contains the string value "null".

This bug seems to affect most rules.

Excel test data is available in the CORE unitTesting/USDM/DDF00098. The results subfolders currently contain results from previous testing showing correct `null` values. If you rerun the rule using the test data, the results contain the incorrect `"null"` values.

**Error returned from Rule Engine**

No error is reported by the rule engine. However, where result values used to look like this:
```json
                    "value": {
                        "parent_entity": "StudyDesign",
                        "parent_id": "StudyDesign_4",
                        "parent_rel": "population",
                        "id": "StudyDesignPopulation_4",
                        "name": "POP4",
                        "plannedSex": null,
                        "plannedSex.decode": null,
                        "id.STUDYCOHORT": null,
                        "name.STUDYCOHORT": null,
                        "plannedSex.STUDYCOHORT": null,
                        "plannedSex.decode.STUDYCOHORT": null
                    },
```
they now look like this:
```json
                    "value": {
                        "parent_entity": "StudyDesign",
                        "parent_id": "StudyDesign_4",
                        "parent_rel": "population",
                        "id": "StudyDesignPopulation_4",
                        "name": "POP4",
                        "plannedSex": "null",
                        "plannedSex.decode": "null",
                        "id.StudyCohort": "null",
                        "name.StudyCohort": "null",
                        "plannedSex.StudyCohort": "null",
                        "plannedSex.decode.StudyCohort": "null"
                    },

```

**Expected behavior**
Missing values should be reported as an unquoted `null` and not as a quoted string, `"null"`.

**Comment by ASL-rmarshall** [comment]
It looks like the "null" values are being assigned [deliberately](https://github.com/cdisc-org/cdisc-rules-engine/blob/0bbccb4f8cd162aec5cd1b54569ab7a67d0a40be/cdisc_rules_engine/models/actions.py#L284).  In this case would it be possible to assign a missing/null value that's compatible with JSON instead of a string value?

Also, lines [287](https://github.com/cdisc-org/cdisc-rules-engine/blob/0bbccb4f8cd162aec5cd1b54569ab7a67d0a40be/cdisc_rules_engine/models/actions.py#L287)-289 make me nervous.  If the value list contains both non-null values and missing/null values, it looks like the whole list would be replaced, meaning that the non-null values would be lost.
